parsel/selector.py

```diff
@@ -399,16 +412,20 @@ def xpath(
 
             selector.xpath('//a[href=$url]', url="http://www.example.com")
         """
-        if self.type == "text":
-            self._load_lxml_root(self.root, type="html")
-        elif self.type not in ("html", "xml"):
+        if self.type not in ("html", "xml", "text"):
             raise ValueError(
                 f"Cannot use xpath on a Selector of type {self.type!r}"
             )
-        try:
-            xpathev = self.root.xpath
-        except AttributeError:
-            return self.selectorlist_cls([])
+        if self.type in ("html", "xml"):
+            try:
+                xpathev = self.root.xpath
+            except AttributeError:
+                return self.selectorlist_cls([])
+        else:
+            try:
+                xpathev = self._get_root(self._text, type="html").xpath
+            except AttributeError:
+                return self.selectorlist_cls([])
```

Comment:
Why is this if/else needed now? Also, I though we were going to consider removing `text` as a type altogether.

Reply:
Well, it is not using `text` per se, it is defaulting to what it can do when there is no type `html` or `xml`, which is the case for str output from jmespath (which doesn't necessarily fits those types).

If you were to use jmespath selector and then go to xpath, the `else` is to one to handle it, because jmespath is sending its result as `text`.
I can be that we may tweak it one more round by removing `text` and use `None`, but afaik `None` can be deceiving if it engulfs `text` scope.

Reply:
I believe the strategy to be used if we are to not lazy load will be to preemptively instantiate/validate `root`, `text`, and so on and let the methods figure out what to do with what they receive given their process' characteristics. 

Centralizing the `type` instantiation on `__init__` becomes hard for the reason we do not know who instantiated it, and what possible outcomes such a previous selector could have.

Reply:
To emphasize and write a better comment... I belive the main issue we are dealing with in this whole endeavor is going from JmesPath to other `etree` selectors. 
This is the reason for the lazy loading which was there previously on this snippet. You would leave JmesPath with `text`, which can be just a raw type (probably the most common case) or with a valid `etree` structure.
If JmesPath was to validate the result it gathered was a valid `etree`, and tweak the child selector to have type `html`, then this provably means we won't ever enter xpath selector with `text` or `None` type, unless it is an incorrect setting which would fail nonetheless.

Summary: If there is a way to add validation of result on JmesPath to make it create new child selectors with `html` type whenever it has a result valid as `etree`, this is probably the next step.